### PR TITLE
[improve][doc] cherry-pick cpp client docs installation section

### PR DESCRIPTION
**Pull request by tisonkun**
Master issue: #17917

See https://github.com/apache/pulsar/issues/17917#issuecomment-1290146425.

### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

cc @momo-jun @Anonymitaet

### CODE REVIEW

**Review by BewareMyPower — APPROVED** [review]

**Review by Anonymitaet — APPROVED** [review]

**Comment by tisonkun** [comment]
@Anonymitaet are there any more required reviewers for this patch?

File: site2/website/versioned_docs/version-2.9.x/client-libraries-cpp.md [hunk 1]
```diff
@@ -5,318 +5,111 @@ sidebar_label: "C++"
...
-
- g++ --std=c++11  PulsarTest.cpp -o test /usr/lib/libpulsar.a -lssl -lcrypto -ldl -lpthread  -I/usr/local/ssl/include -L/usr/local/ssl/lib -lboost_system -lboost_regex -lcurl -lprotobuf -lzstd -lz
-
+wget @pulsar:dist_rpm:client-debuginfo@
 ```
 
-4. `libpulsarwithdeps.a` is a static library, based on `libpulsar.a`. It is archived in the dependencies of `libboost_regex`, `libboost_system`, `libcurl`, `libprotobuf`, `libzstd` and `libz`. You can use this Pulsar library with the command below.
-
-```bash
+This package contains debug symbols for `libpulsar.so`
```
Comment (momo-jun) [inline site2/website/versioned_docs/version-2.9.x/client-libraries-cpp.md:82]:
```suggestion
This package contains debug symbols for `libpulsar.so`.
```

File: site2/website/versioned_docs/version-2.8.x/client-libraries-cpp.md [hunk 2]
```diff
@@ -5,267 +5,111 @@ sidebar_label: "C++"
...
-| pulsar-client-dev | Static library `libpulsar.a`, `libpulsarwithdeps.a` and C++ and C headers |
-
-## MacOS
-
-### Compilation
-
-1. Clone the Pulsar repository.
-
-```shell
+This package contains debug symbols for `libpulsar.so`
```
Comment (momo-jun) [inline site2/website/versioned_docs/version-2.8.x/client-libraries-cpp.md:82]:
```suggestion
This package contains debug symbols for `libpulsar.so`.
```

File: site2/website/versioned_docs/version-2.7.5/client-libraries-cpp.md [hunk 3]
```diff
@@ -5,161 +5,111 @@ sidebar_label: "C++"
...
-pulsar-client-cpp/pkg/rpm/docker-build-rpm.sh
+</TabItem>
+<TabItem value="client-debuginfo">
 
+```bash
+wget @pulsar:dist_rpm:client-debuginfo@
 ```
 
-This builds the RPM inside a Docker container and it leaves the RPMs in `pulsar-client-cpp/pkg/rpm/RPMS/x86_64/`.
+This package contains debug symbols for `libpulsar.so`
```
Comment (momo-jun) [inline site2/website/versioned_docs/version-2.7.5/client-libraries-cpp.md:82]:
```suggestion
This package contains debug symbols for `libpulsar.so`.
```

File: site2/website/versioned_docs/version-2.10.x/client-libraries-cpp.md [hunk 4]
```diff
@@ -5,341 +5,111 @@ sidebar_label: "C++"
...
+```bash
+wget @pulsar:dist_rpm:client-debuginfo@
 ```
 
-## Windows (64-bit)
-
-### Compilation
-
-1. Clone the Pulsar repository.
+This package contains debug symbols for `libpulsar.so`
```
Comment (momo-jun) [inline site2/website/versioned_docs/version-2.10.x/client-libraries-cpp.md:82]:
```suggestion
This package contains debug symbols for `libpulsar.so`.
```

**Review by momo-jun — APPROVED** [review]
LGTM.

**Comment by Anonymitaet** [comment]
> @Anonymitaet are there any more required reviewers for this patch?

@tisonkun I guess no since we already have eyes on both technical and tw perspectives.

**Pull request merged**